share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -736,7 +736,7 @@ deprecated-ci-build:
       - cd "${CI_PROJECT_DIR}" && curl -sOL https://raw.githubusercontent.com/spack/spack-configs/main/AWS/parallelcluster/postinstall.sh
       - sed -i -e "s/spack arch -t/echo ${SPACK_TARGET_ARCH}/g" postinstall.sh
       - /bin/bash postinstall.sh -fg
-      - spack config --scope site add "packages:all:target:\"target=${SPACK_TARGET_ARCH}\""
+      - spack config --scope site add "packages:all:target:\"${SPACK_TARGET_ARCH}\""
```

Comment:
Is this working? Package preferences expect an array of strings for targets, and this is a single target. Trying the same from the command line I get an error:
```console
$ spack config add packages:all:target:x86_64
==> Error: <unicode string>:1: 'x86_64' is not of type 'array'
```

Reply:
Yes, it works for me. Maybe it's the escaped quotes? 
DSo you are saying the correct syntax would be `spack config --scope site add "packages:all:target:[\
"${SPACK_TARGET_ARCH}\"]"`?

Reply:
Can you post the resulting YAML file?

Reply:
```
bash-4.2# spack config --scope site add "packages:all:target:\"target=${SPACK_TARGET_ARCH}\""
bash-4.2# cat etc/spack/packages.yaml
packages:
  [...]
  all:
    compiler: [nvhpc, gcc, clang]
    providers:
      blas: [armpl-gcc, openblas]
      fftw-api: [armpl-gcc, fftw]
      lapack: [armpl-gcc, openblas]
      mpi: [openmpi, mpich]
      scalapack: [netlib-scalapack]
    permissions:
      read: world
      write: user
    target:
    - target=neoverse_n1
```

Reply:
Ok, I'd like to get to the bottom of this:
![Screenshot from 2023-06-08 15-17-25](https://github.com/spack/spack/assets/4199709/80026d0c-ce4c-4343-af6b-c11a2128b356)

I am using:

* **Spack:** 0.21.0.dev0 (315873cbd329a05f8cb944bcfb8c9ec2866493ff)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-icelake
* **Concretizer:** clingo

Reply:
Confirmed that in:
```console
docker run -it --rm "ghcr.io/spack/pcluster-amazonlinux-2:v2023-05-25"
```
it works as reported by @stephenmsachs :thinking: Need to check if it is the Python version being used (3.7 vs. 3.8). In any case:
```console
$ spack config --scope site add packages:all:target:"[${SPACK_TARGET_ARCH}]"
```
works on both platforms.